subsys/pm/device.c

```diff
@@ -67,16 +67,21 @@ int pm_device_action_run(const struct device *dev,
 		/*
 		 * TURN_ON and TURN_OFF are actions triggered by a power domain
 		 * when it is resumed or suspended, which means that the energy
-		 * to the device will be removed or added. For this reason, even
-		 * if the device does not handle these actions its state needs to
-		 * updated to reflect its physical behavior.
+		 * to the device will be removed or added. For this reason, if
+		 * the transition fails or the device does not handle these
+		 * actions its state still needs to updated to reflect its
+		 * physical behavior.
 		 *
 		 * The function will still return the error code so the domain
 		 * can take whatever action is more appropriated.
 		 */
-		if ((ret == -ENOTSUP) && ((action == PM_DEVICE_ACTION_TURN_ON)
-				  || (action == PM_DEVICE_ACTION_TURN_OFF))) {
+		switch (action) {
+		case PM_DEVICE_ACTION_TURN_ON:
```

Comment:
nit: missing `__fallthrough;`

Reply:
Is this required in the situation where there are no statements in the case?
My understanding was that it was only needed when it might look like there is a missing `break`.

Reply:
I can't find any docs, but I see many cases like this where `__fallthrough` is used. @stephanosio @nashif ?

Reply:
`__fallthrough` aka. `__attributte__(fallthrough)` is not required in this case because this can be seen as "a case with multiple conditions" and not necessarily an implicit fallthrough. For this reason, `-Wimplicit-fallthrough` does not complain about such usages.